sky/clouds/lambda_cloud.py

```diff
@@ -0,0 +1,245 @@
+                    accelerators=None,
+                )
+                resource_list.append(r)
+            return resource_list
+
+        # Currently, handle a filter on accelerators only.
+        accelerators = resources.accelerators
+        if accelerators is None:
+            # No requirements to filter, so just return a default VM type.
+            return (_make([Lambda.get_default_instance_type()]),
+                    fuzzy_candidate_list)
+
+        assert len(accelerators) == 1, resources
+        acc, acc_count = list(accelerators.items())[0]
+        (instance_list, fuzzy_candidate_list
+        ) = service_catalog.get_instance_type_for_accelerator(
+            acc,
+            acc_count,
+            use_spot=resources.use_spot,
+            region=resources.region,
+            zone=resources.zone,
+            clouds='lambda')
+        if instance_list is None:
+            return ([], fuzzy_candidate_list)
+        return (_make(instance_list), fuzzy_candidate_list)
+
+    def check_credentials(self) -> Tuple[bool, Optional[str]]:
+        try:
+            lambda_utils.LambdaCloudClient().list_instances()
+        except (AssertionError, KeyError, lambda_utils.LambdaCloudError):
+            return False, ('Failed to access Lambda Cloud with credentials. '
+                           'To configure credentials, go to:\n    '
+                           '  https://cloud.lambdalabs.com/api-keys\n    '
+                           'to generate API key and add the line\n    '
+                           '  api_key = [YOUR API KEY]\n    '
+                           'to ~/.lambda_cloud/lambda_keys')
+        return True, None
+
+    def get_credential_file_mounts(self) -> Dict[str, str]:
+        return {
+            f'~/.lambda_cloud/{filename}': f'~/.lambda_cloud/{filename}'
+            for filename in _CREDENTIAL_FILES
+        }
+
+    def instance_type_exists(self, instance_type: str) -> bool:
+        return service_catalog.instance_type_exists(instance_type, 'lambda')
+
+    def validate_region_zone(self, region: Optional[str], zone: Optional[str]):
+        return service_catalog.validate_region_zone(region,
+                                                    zone,
+                                                    clouds='lambda')
+
+    def accelerator_in_region_or_zone(self,
+                                      accelerator: str,
+                                      acc_count: int,
+                                      region: Optional[str] = None,
+                                      zone: Optional[str] = None) -> bool:
+        return service_catalog.accelerator_in_region_or_zone(
+            accelerator, acc_count, region, zone, 'lambda')
+
+    @classmethod
+    def supports(
+            cls, requested_features: Set[clouds.CloudImplementationFeatures]
+    ) -> bool:
+        return requested_features.issubset(_LAMBDA_IMPLEMENTATION_FEATURES)
```

Comment:
Can we have a TODO here for `get_current_user_identity`? We may want to add that function for Lambda as well, to avoid the cluster being accidently removed from the cluster table when the user switch their API key.

Reply:
Added!